skipchain/protocol.go

```diff
@@ -88,17 +87,25 @@ func NewProtocolGetBlocks(n *onet.TreeNodeInstance) (onet.ProtocolInstance, erro
 // Start sends the extend roster request to all of the children.
 func (p *ExtendRoster) Start() error {
 	log.Lvl3("Starting Protocol ExtendRoster")
-	errs := p.SendToChildrenInParallel(p.ExtendRoster)
-	if len(errs) > p.allowedFailures {
-		return fmt.Errorf("Send to children failed: %v", errs)
-	}
+	go func() {
```

Comment:
When I see this, I think of Dave Cheny, who says, "always know how your goroutine will exit before you start it". If p.SendToChildrenInParallel blocks for some reason, this goroutine will never exit. tcp.go uses read deadlines, but not write deadlines. It seems likely that an attempt to send to a peer who is accepting TCP but not talking (or has wedged and stops talking) will result in a wedged goroutine.

But that problem is not related to what you are trying to solve here, so feel free to just file an issue and kick it down the road a bit.

Reply:
Opened dedis/onet#416